### DISCUSSION TITLE
Disable DKIM / SPF or whitelist network

### DISCUSSION THREAD

**Discussion by taumeister** [rspamd/rspamd#4363] [Q&A]
Hello Community,

I have a working constellation of Mailcow (integrated rspamd) with Ciphermail encryption gateway.
Mailcow uses Postfix's 'content_filter' and sends Mails to Ciphermail ( which is on the same network ) and Ciphermail sends back to Mailcow via port 10026 after en- or decryption.
So far, so good. However, since I connected Ciphermail, the DKIM check no longer works, or is invalid.

Here is my question: Is it possible to ignore DKIM and SPF checks from one server, maybe to 'whitelist' it?
I skimmed the documentation, it didn't become clear to me exactly.

I read something about whitelisted networks, but don't know how to configure it.
[https://rspamd.com/doc/modules/dkim.html](url)
..whitelist (map): map of whitelisted networks

Can someone help me with this, possibly provide an example?

For you help thanks in advance
Greetings
Thomas

**Comment by fatalbanana** [comment]
It's an IP map, see here for hints about how those are configured & populated:

https://rspamd.com/doc/faq.html#what-are-maps
https://rspamd.com/doc/faq.html#what-can-be-in-the-maps

SPF module similarly has a setting called `whitelist`.

More generally you can disable any symbol by name via Lua, in this case the correct symbols to be disabled are called `SPF_CHECK` and `DKIM_CHECK`. See here: https://rspamd.com/doc/faq.html#how-can-i-disable-some-rspamd-rules-safely

Settings module is another possibility: https://rspamd.com/doc/configuration/settings.html

**Comment by taumeister** [comment]
First of all, thank you so much for answering me.

I understood that I can create some kind of ruleset with so called maps that have key/value entries. 
I didn't really understand what 'symbol' means in this context and I don't know lua either...phew.

Actually I just want that when Mailcow, with integrated rspamd passes mails to my gateway for encryption and it gets back the mails afterwards that DKIM and SPF are still ok.
Is whitelisting and mapping the right method for this?
I'm not an rspamd expert unfortunately...

Would it be possible to tell me which file to create or modify so that I can reach my goal? Possibly a concrete example?

That would be great, thank you
Thomas

**Reply by fatalbanana** [reply]
A symbol is any kind of check and/or result performed by Rspamd (the likes of `R_SPF_FAIL` which you would see in the result).

If you can't tolerate that SPF/DKIM are broken, disable the checks (probably conditionally).

If you don't want them to be broken, maybe there is something you can do about that (like signing the messages later).

If you want to pretend everything is OK when it isn't that is possible (you could remove symbols and add new ones after the fact in Lua, or similarly disable the check & insert a fake result early on).

**Comment by taumeister** [comment]
What I have seen is a tutorial from here: 'https://community.nethserver.org/t/rspamd-allow-ip-whitelist/11807'.
I have already an IP_Whitelist entry in .../locald.d/multimap.conf, but if I understand correctly, I only control the score with it.
How do I control DKIM stuff?
